src/main/java/de/tum/in/www1/artemis/config/MetricsBean.java

```diff
@@ -113,13 +165,231 @@ private void registerWebsocketMetrics() {
+    }
+
+    private double getUpcomingStartingExamCount(int minutes) {
+        SecurityUtils.setAuthorizationObject();
+        var now = ZonedDateTime.now();
+        var endDate = ZonedDateTime.now().plusMinutes(minutes);
+        return examRepository.countExamsWithStartDateGreaterThanEqualButLessOrEqualThan(now, endDate);
+    }
+
+    private double getUpcomingStartingExamCountWithStudentMultiplier(int minutes) {
+        SecurityUtils.setAuthorizationObject();
+        var now = ZonedDateTime.now();
+        var endDate = ZonedDateTime.now().plusMinutes(minutes);
+        return examRepository.countExamUsersInExamsWithStartDateGreaterThanEqualButLessOrEqualThan(now, endDate);
+    }
+
+    @Scheduled(fixedRate = 15 * 60 * 1000, initialDelay = 0) // Every 15 minutes
+    public void updatePublicArtemisMetrics() {
+        // The authorization object has to be set because this method is not called by a user but by the scheduler
+        SecurityUtils.setAuthorizationObject();
+
+        ZonedDateTime now = ZonedDateTime.now();
+
+        var activeUserPeriodsInDays = new Integer[] { 1, 7, 14, 30 };
+        activeUserMultiGauge.register(Stream.of(activeUserPeriodsInDays)
+                .map(periodInDays -> MultiGauge.Row.of(Tags.of("period", periodInDays.toString()), statisticsRepository.countActiveUsers(now.minusDays(periodInDays), now)))
+                .collect(Collectors.toCollection(ArrayList::new)), true);
+
+        var courses = courseRepository.findAll();
+        // We set the number of students once to prevent multiple queries for the same date
+        courses.forEach(course -> course.setNumberOfStudents((long) userRepository.getStudents(course).size()));
+
+        courses.forEach(course -> {
+            if (course.getSemester() == null) {
+                course.setSemester("No semester");
+            }
+            if (course.getTitle() == null) {
+                if (course.getShortName() != null) {
+                    course.setTitle("Course" + course.getShortName());
+                }
+                else {
+                    course.setTitle("Course" + course.getId().toString());
+                }
+            }
+        });
+
+        var activeCourses = courses.stream()
+                .filter(course -> (course.getStartDate() == null || course.getStartDate().isBefore(now)) && (course.getEndDate() == null || course.getEndDate().isAfter(now)))
+                .toList();
+
+        studentsCourseGauge.register(
+                activeCourses.stream().map(course -> MultiGauge.Row.of(Tags.of("courseName", course.getTitle(), "semester", course.getSemester()), course.getNumberOfStudents()))
+                        .collect(Collectors.toCollection(ArrayList::new)),
+                true);
+
+        activeCoursesGauge.set(activeCourses.size());
+        coursesGauge.set(courseRepository.findAll().size());
```

Comment:
You already have got `var courses` above.
```suggestion
        coursesGauge.set(courses.size());
```